### DISCUSSION TITLE
Airflow does not register DAG unless it reads the keyword "airflow" in the python file

### DISCUSSION THREAD

**Discussion by dhawal-rathore** [apache/airflow#39803] [General]
### Apache Airflow version

2.9.1

### What happened?

I am extending the DAG class in a class file, and importing it and creating an object did not list my DAG, but creating a DAG object worked. So after a while I figured out the scheduler does not add the DAG if it does not detect the string "airflow" inside the code. 

### What you think should happen instead?

It should go through the code to see if a DAG object exists in the code, or this should be mentioned in the documentation.

### How to reproduce

1. Make a class file and a class which extends the DAG class, should not have airflow in the name.
2. Create the DAG file in your DAG folder, and import the class into the file. Make sure the word "airflow" is not in the file.
3. Your DAG will not be registered

### Operating System

Ubuntu

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
This is intended behaviour, see Configuration Reference: [`[core] dag_discovery_safe_mode`](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-discovery-safe-mode)
